graphics/font.cpp

```diff
@@ -508,4 +508,70 @@ TextAlign convertTextAlignH(TextAlign alignH, bool rtl) {
 	}
 }
 
+#define wholedivide(x, y)	(((x)+((y)-1))/(y))
+
+static void countupScore(int *dstGray, int x, int y, int bbw, int bbh, float scale) {
+	int newbbw = bbw * scale;
+	int newbbh = bbh * scale;
+	int x_ = x * newbbw;
+	int y_ = y * newbbh;
+	int x1 = x_ + newbbw;
+	int y1 = y_ + newbbh;
+
+	int newxbegin = x_ / bbw;
+	int newybegin = y_ / bbh;
+	int newxend = wholedivide(x1, bbw);
+	int newyend = wholedivide(y1, bbh);
+
+	for (int newy = newybegin; newy < newyend; newy++) {
+		for (int newx = newxbegin; newx < newxend; newx++) {
+			int newX = newx * bbw;
+			int newY = newy * bbh;
+			int newX1 = newX + bbw;
+			int newY1 = newY + bbh;
+			dstGray[newy * newbbw + newx] += (MIN(x1, newX1) - MAX(x_, newX)) *
+											 (MIN(y1, newY1) - MAX(y_, newY));
+		}
+	}
+}
+
+static void magnifyGray(Surface *src, int *dstGray, int width, int height, float scale) {
+	for (uint16 y = 0; y < height; y++) {
+		for (uint16 x = 0; x < width; x++) {
+			if (*((byte *)src->getBasePtr(x, y)) == 1)
+				countupScore(dstGray, x, y, width, height, scale);
+		}
+	}
+}
+
+void Font::initializeScaling(int srcSize, int newSize) {
+	scaleSurface = new Surface();
+	scaleSurface->create(MAX(srcSize * 2, newSize * 2), MAX(srcSize * 2, newSize * 2),PixelFormat::createFormatCLUT8());
+	grayScaleMapSize = newSize * 20 * newSize;
+	grayScaleMap = (int *)malloc(grayScaleMapSize * sizeof(int));
+}
+
+Surface *Font::scaleSingleGlyph(int width, int height, int grayLevel, int chr, int xOffset, int srcheight, int srcwidth, float scale){
```

Comment:
Bad code formatting

Reply:
I think it is the function line, please check I have changed it